buildSrc/src/main/java/com/google/firebase/gradle/plugins/DackkaPlugin.md

```diff
@@ -0,0 +1,99 @@
+# Dackka Plugin
+
+If the implementation of the DackkaPlugin is not straight forward enough
+for you to understand, hopefully this document will help you.
+
+## Dackka
+
+Dackka is an internal-purposed Dokka plugin. Google hosts its documentation on
+an internal service called **Devsite**. Firebase hosts their documentation on a
+variant of Devsite called **Firesite**. You can click [here](https://firebase.google.com/docs/reference) 
+to see how that looks. Essentially, it's just Google's way of decorating (and organizing)
+documentation.
+
+Devsite expects its files to be in a very specific format. Previously, we would
+use an internal variant of metalava called doclava- which allowed us to provide
+sensible defaults as to how the Javadoc should be rendered. Then, we would do
+some further transformations to get the Javadoc output in-line with what
+Devsite expects. This was a lengthy process, and came with a lot of overhead.
+
+Dackka is an internal solution to that. Dackka provides a devsite plugin for
+Dokka that will handle the job of doclava. Not only does this mean we can cut
+out a huge portion of our transformation systems- but the overhead for maintaining
```

Comment:
Which systems are you referring to? iiuc we will still have to run our source through metalava for example

Reply:
We shouldn't have to, not with dackka at least. There's a diagram provided in the docs for the plug-in, but basically dackka handles all that metalava previously provided. Unless I'm missing something?

Reply:
hm, does metalava remove hidden superclasses for example?

Reply:
define "remove". It doesnt generate documentation for the class, but its still referenced as a parent. It also seems to still generate a link for it.

Reply:
Although, it seems thats a bug. b/243324828

Reply:
> but its still referenced as a parent

This is not the behavior we get with metalava afaicr, ideally we should preserve current behavior, i.e.:

Source:
```java
/** @hide */ class Base {}
class Derived extends Base {}
```

Docstubs with metalava result:
```java
class Derived {}
```

One example of this that I found is [FirebaseInAppMessagingDisplay](https://firebase.devsite.corp.google.com/docs/reference/kotlin/com/google/firebase/inappmessaging/display/FirebaseInAppMessagingDisplay?hl=en)

Reply:
Ahh I see. Owen does say that that's the behavior they want, but its not current in place per that bug. For the time being, do you just wanna have dackka pull the sources from doc-stubs?